app/src/androidTest/java/org/mozilla/focus/activity/AddtoHSTest.java

```diff
@@ -0,0 +1,239 @@
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("image_test.html")));
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("rabbit.jpg")));
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("download.jpg")));
+
+                webServer.start();
+            } catch (IOException e) {
+                throw new AssertionError("Could not start web server", e);
+            }
+        }
+
+        @Override
+        protected void afterActivityFinished() {
+            super.afterActivityFinished();
+
+            try {
+                webServer.close();
+                webServer.shutdown();
+            } catch (IOException e) {
+                throw new AssertionError("Could not stop web server", e);
+            }
+        }
+    };
+
+    private UiObject titleMsg = TestHelper.mDevice.findObject(new UiSelector()
+            .description("focus test page")
+            .enabled(true));
+    UiObject welcomeBtn = TestHelper.mDevice.findObject(new UiSelector()
+            .resourceId("com.android.launcher3:id/cling_dismiss_longpress_info")
+            .text("GOT IT")
+            .enabled(true));
+
+    void removeWelcomeOverlay() throws UiObjectNotFoundException {
+        final String FOCUS_DEBUG_APP = "org.mozilla.focus.debug";
+
+        // Note: in case of some older simulators, the Welcome overlay covers
+        // where the shortcut icons should be!
+        TestHelper.pressHomeKey();
+        if (welcomeBtn.exists()) {
+            welcomeBtn.click();
+        }
+        // Re-Launch the app
+        Context context = InstrumentationRegistry.getInstrumentation()
+                .getTargetContext()
+                .getApplicationContext();
+        final Intent intent = context.getPackageManager()
+                .getLaunchIntentForPackage(FOCUS_DEBUG_APP);
+        context.startActivity(intent);
+    }
+
+    @Test
+    public void AddToHomeScreenTest() throws InterruptedException, UiObjectNotFoundException, IOException {
+
+        UiObject shortcutIcon = TestHelper.mDevice.findObject(new UiSelector()
+                .className("android.widget.TextView")
+                .description("For Testing Purpose")
+                .enabled(true));
+
+        removeWelcomeOverlay();
+
+        // Open website, and click 'Add to homescreen'
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.clearTextField();
+        TestHelper.inlineAutocompleteEditText.setText(webServer.url(TEST_PATH).toString());
+        TestHelper.hint.waitForExists(waitingTime);
+        TestHelper.pressEnterKey();
+        TestHelper.webView.waitForExists(waitingTime);
+        Assert.assertTrue("Website title loaded", titleMsg.exists());
```

Comment:
\[PMD\] INFO: The String literal "Website title loaded" appears 4 times in this file; the first occurrence is on line 132